test/allof.test.js

```diff
@@ -553,50 +553,165 @@ test('allof with local anchor reference', (t) => {
   t.equal(stringify(data), JSON.stringify(data))
 })
 
+test('allOf: multiple nested $ref properties', (t) => {
+  t.plan(2)
+
+  const externalSchema1 = {
+    $id: 'externalSchema1',
+    oneOf: [
+      { $ref: '#/definitions/id1' }
+    ],
+    definitions: {
+      id1: {
+        type: 'object',
+        properties: {
+          id1: {
+            type: 'integer'
+          }
+        },
+        additionalProperties: false
+      }
+    }
+  }
+
+  const externalSchema2 = {
+    $id: 'externalSchema2',
+    oneOf: [
+      { $ref: '#/definitions/id2' }
+    ],
+    definitions: {
+      id2: {
+        type: 'object',
+        properties: {
+          id2: {
+            type: 'integer'
+          }
+        },
+        additionalProperties: false
+      }
+    }
+  }
+
+  const schema = {
+    anyOf: [
+      { $ref: 'externalSchema1' },
+      { $ref: 'externalSchema2' }
+    ]
+  }
+
+  const stringify = build(schema, { schema: [externalSchema1, externalSchema2] })
+
+  t.equal(stringify({ id1: 1 }), JSON.stringify({ id1: 1 }))
+  t.equal(stringify({ id2: 2 }), JSON.stringify({ id2: 2 }))
+})
+
 test('allOf: throw Error if types mismatch ', (t) => {
-  t.plan(1)
+  t.plan(3)
 
   const schema = {
     allOf: [
       { type: 'string' },
       { type: 'number' }
     ]
   }
-  t.throws(() => build(schema), new Error('allOf schemas have different type values'))
+  try {
+    build(schema)
+  } catch (error) {
+    t.ok(error instanceof Error)
+    t.equal(error.message, 'Failed to merge "type" keyword schemas.')
+    t.same(error.schemas, [['string'], ['number']])
+  }
 })
 
 test('allOf: throw Error if format mismatch ', (t) => {
-  t.plan(1)
+  t.plan(3)
 
   const schema = {
     allOf: [
       { format: 'date' },
       { format: 'time' }
     ]
   }
-  t.throws(() => build(schema), new Error('allOf schemas have different format values'))
+  try {
+    build(schema)
+  } catch (error) {
+    t.ok(error instanceof Error)
+    t.equal(error.message, 'Failed to merge "format" keyword schemas.')
+    t.same(error.schemas, ['date', 'time'])
+  }
 })
 
-test('allOf: throw Error if nullable mismatch /1', (t) => {
+test('recursive nested allOfs', (t) => {
```

Comment:
Why did you change these two allOf tests?

Reply:
I didn't change the test here. I removed `allOf: throw Error if nullable mismatch /1` test and added a couple of new tests. I dropped the `allOf: throw Error if nullable mismatch /1` because now the merging algoritm is more accurate and can resolve the situation when two schemas has different `nullable` keyword values.

If one schema has `nullable` = true and another one has `nullable` = false the merged schema will have `nullable` = false. Now `fast-json-stringify` just throws an error in this case.